### PR TITLE
TETP optimize youth event list loading

### PR DESCRIPTION
**Pull request by pekkvil**

### CODE REVIEW

**Review by nenonja — COMMENTED** [review]
Thinking about the JobPostings, you could use `useInifiniteQuery` hook from `react-query` ? It would give the `hasMore` boolean for loading more button and other functionality out of the box: https://react-query.tanstack.com/guides/infinite-queries
Another option is Paginated Queries: https://react-query.tanstack.com/guides/paginated-queries 

Not sure which one suits better but I think it would be much easier to use them instead of implementing custom pagination and loading logic?

**Comment by pekkvil** [comment]
> 

`useInifiniteQuery` with pagination was used earlier which worked really nicely. Problem was that when we added map to show all the locactions for TET places, there was a need to fetch all the TET places at once. It didn't feel right to keep useInfinitQuery functionality and also make a request that fetches all. Fetching them all still takes some time so current solution is to also fetch the first page and allow use for the pagination when request for all the places is finished (pagination is then done on the client side by just slicing array).

File: frontend/tet/shared/src/components/map/Map.tsx [hunk 1]
```diff
@@ -86,9 +96,19 @@ const Map: React.FC<Props> = ({
                 <$Date>{getDateString(posting)}</$Date>
                 <$Address>{getAddressString(posting)}</$Address>
                 {showLink && (
-                  <Link href={`/postings/show?id=${posting.id}`} size="L">
-                    {t(`common:map.readMore`)}
-                  </Link>
+                  <Button
```
[the remark this answers is not in the document]
Comment (pekkvil) [reply frontend/tet/shared/src/components/map/Map.tsx:99]: Thanks! I'll give that a try

File: frontend/tet/admin/src/styles.css [hunk 2]
```diff
@@ -0,0 +1,4 @@
+/* Use until implemented to hds */
+div#hds-tag {
```
[the remark this answers is not in the document]
Comment (pekkvil) [reply frontend/tet/admin/src/styles.css:2]: This would also change tag's border on kesäseteli's side?

**Comment by nenonja** [comment]
> > 
> 
> `useInifiniteQuery` with pagination was used earlier which worked really nicely. Problem was that when we added map to show all the locactions for TET places, there was a need to fetch all the TET places at once. It didn't feel right to keep useInfinitQuery functionality and also make a request that fetches all. Fetching them all still takes some time so current solution is to also fetch the first page and allow use for the pagination when request for all the places is finished (pagination is then done on the client side by just slicing array).

Yes then it is okay if you have already tried, just wanted to know that you are aware of those queries in react-query 👍

**Review by mattikl — COMMENTED** [review]
Tested in the review environment.

**Review by mattikl — APPROVED** [review]

**Pull request merged**